### ISSUE TITLE
[Feature]: Split main & debug image building CI steps

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger#4471]
### Requirement

I want CI to be as fast as possible

### Problem

I just observed that [Build all-in-one](https://github.com/jaegertracing/jaeger/actions/runs/5093713722/jobs/9156660775?pr=4460) was stuck for 7min on installing delve debugger into the base image:
```
Fri, 26 May 2023 18:10:51 GMT #21 [linux/arm64 build 3/3] RUN if [[ "arm64" == "s390x"  ||  "arm64" == "ppc64le" ]] ; then 	touch /go/bin/dlv;     else         go install github.com/go-delve/delve/cmd/dlv@latest;     fi
Fri, 26 May 2023 18:17:14 GMT #21 DONE 442.7s
```

### Proposal

Building debug images is only needed when we do an official release, they do not provide any extra signal during CI for PRs. I suggest we adjust the build scripts to skip building debug versions unless the GH actions runs for a release.

**Comment by psk001** [comment]
Hi @yurishkuro , I would like to go ahead with this issue. How do I proceed?

**Comment by yurishkuro** [comment]
* check .github/workflows/ci-all-in-one-build.yml and ci-docker-build.yml workflows, find which scripts are invoked
* update the script to take an extra param like SKIP_DEBUG and bypass debug image building

**Comment by psk001** [comment]
Thanks for the heads up , I will proceed accordingly.

**Comment by wasup-yash** [comment]
> Thanks for the heads up, I will proceed accordingly.

hey, just wanna know if you are working on this issue? and if not i would like to take this up

**Comment by yurishkuro** [comment]
This is solved by #4496
